### PR TITLE
sol

### PR DESCRIPTION
**Pull request by yurii-krymskiy**
[DEMO LINK](https://yurii-krymskiy.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,175 @@
...
+      );
+        break;
+      default:
+        break;
+    }
+
+    return visibleTodos;
+  };
+
+  const handleInput = (event: any) => {
```
Comment (mykhalenych) [inline src/App.tsx:63]: don't use any in typescript, add types

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,175 @@
...
+    setNewTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = async (event: any) => {
```
Comment (mykhalenych) [inline src/App.tsx:67]: check all any

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,175 @@
...
+    if (newTodoTitle.trim()) {
+      try {
+        setLoading(true);
+        const newTodoResponse = await addNewTodo(newTodo);
+
+        setTodos((prevTodos) => [...prevTodos, newTodoResponse]);
+        setTempTodo(null);
+      } catch (error) {
+        setErrorMessage(Error.Loading);
+        setErrorMessage(Error.Adding);
```
Comment (mykhalenych) [inline src/App.tsx:95]: why do you set two messages?

File: src/App.tsx [hunk 4]
```diff
@@ -1,24 +1,175 @@
...
+  };
+
+  const handleDelete = async (id: number) => {
+    try {
+      setLoading(true);
+      await deleteTodo(id);
+      setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== id));
+    } catch (error) {
+      setErrorMessage(Error.Loading);
+      setErrorMessage(Error.Deleting);
```
Comment (mykhalenych) [inline src/App.tsx:110]: the same

File: src/components/Notification.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  errorMessage: Error ;
+  showNotification: boolean
+  setShowNotification: (showNotification: boolean)=> void;
+};
+
+export const Notification: FC <Props> = ({
+  errorMessage, showNotification, setShowNotification,
+}) => {
+  useEffect(() => {
+    if (showNotification) {
```
Comment (mykhalenych) [inline src/components/Notification.tsx:15]: it is reduntand state you can check if(errorMessage !== Error.NoError)

File: src/components/Notification.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      const timer = setTimeout(() => {
+        setShowNotification(false);
```
Comment (mykhalenych) [inline src/components/Notification.tsx:17]: and set default error message

File: src/components/Notification.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    } return;
```
Comment (mykhalenych) [inline src/components/Notification.tsx:21]:
```suggestion
    }
```

File: src/components/Notification.tsx [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  }, [showNotification]);
+
+  return showNotification ? (
```
Comment (mykhalenych) [inline src/components/Notification.tsx:24]:
```suggestion
  return showNotification && (
```

File: src/components/Notification.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    <div className="notification is-danger is-light has-text-weight-normal">
+      <button
+        type="button"
+        className="delete"
+        aria-label="Mute volume"
+        onClick={() => setShowNotification(false)}
+      />
+      {errorMessage}
+    </div>
+  ) : (
```
Comment (mykhalenych) [inline src/components/Notification.tsx:34]:
```suggestion
  )
```

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 10]
```diff
@@ -1,24 +1,175 @@
...
+    } catch (error) {
+      setErrorMessage(Error.Loading);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos();
+    console.log(tempTodo);
```
Comment (volodymyr-soltys97) [inline src/App.tsx:39]: Do you need `console.log()` here?

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Let's improve your code

File: src/App.tsx [hunk 11]
```diff
@@ -1,24 +1,174 @@
...
 
-const USER_ID = 0;
+const USER_ID = 6752;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterValue, setFilterValue] = useState('all');
+  const [errorMessage, setErrorMessage] = useState<Error>(Error.NoError);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [showNotification, setShowNotification] = useState<boolean>(false);
```
Comment (mykhalenych) [inline src/App.tsx:19]: delete this state, and check if errorMessage === Error.NoError, and when you need to clear state use setErrorMessage(Error.NoError)

**Review by mykhalenych — CHANGES_REQUESTED** [review]

**Pull request closed without merging**